src/Propel/Runtime/ActiveQuery/Criteria.php

```diff
@@ -2867,22 +2850,41 @@ public function doSelect(?ConnectionInterface $con = null)
 
         $params = [];
         $sql = $this->createSelectSql($params);
+        $stmt = null;
         try {
             $stmt = $con->prepare($sql);
             $db->bindValues($stmt, $params, $dbMap);
             $stmt->execute();
         } catch (Exception $e) {
-            if (isset($stmt)) {
-                $stmt = null; // close
-            }
-            Propel::log($e->getMessage(), Propel::LOG_ERR);
-
-            throw new PropelException(sprintf('Unable to execute SELECT statement [%s]', $sql), null, $e);
+            $this->handleStatementException($e, $stmt, $sql);
         }
 
         return $con->getDataFetcher($stmt);
     }
 
+    /**
+     * Logs an exception and adds the complete SQL statement to the exception.
+     *
+     * @param \Exception $e
+     * @param \Propel\Runtime\Connection\StatementInterface|\PDOStatement $stmt
+     * @param string|null $sql
+     *
+     * @throws \Propel\Runtime\Exception\PropelException
+     *
+     * @return void
+     */
+    protected function handleStatementException(Exception $e, $stmt, ?string $sql = '')
```

Comment:
If you add a new method, we could add `: void` here now.

Reply:
Added return type. Thank you for finding!